pkg/services/ngalert/api/api_ruler.go

```diff
@@ -511,6 +606,40 @@ func (c *changes) isEmpty() bool {
 	return len(c.Update)+len(c.New)+len(c.Delete) == 0
 }
 
+// mergeChanges will return a new change set that contains all changes
+// that are in set a and set b.
+func mergeChanges(a, b *changes) *changes {
```

Comment:
```suggestion
// intersectChanges will return a new change set that contains all changes
// that are in both set a and set b.
func intersectChanges(a, b *changes) *changes {
```
I think it's important to document that this is the intersection of a and b, not the union. I originally thought this was trying to find the union (add the changes) not the shared ones.

Reply:
Changed it 👍